extensions/lifecycle/lifecycleProducer/LifecycleProducer.js

```diff
@@ -0,0 +1,377 @@
+        const { bucket, owner } = result.value;
+        return async.waterfall([
+            next => this._getAccountCredentials(owner, next),
+            (accountCreds, next) => {
+                const s3 = this._getS3Client(accountCreds);
+                const params = { Bucket: bucket };
+                return s3.getBucketLifecycleConfiguration(params, next);
+            },
+        ], (err, config) => {
+            if (err) {
+                this._log.error('error getting bucket lifecycle config', {
+                    error: err,
+                });
+                return cb(err);
+            }
+            if (!this._shouldProcessConfig(config)) {
+                return cb();
+            }
+            this._log.info('scheduling new task for bucket lifecycle', {
+                bucket,
+                method: 'LifecycleProducer._processBucketEntry',
+            });
+            return this._internalTaskScheduler.push({
+                // TODO: Pass new `LifecycleTask(this)` here once implemented.
+                task: { processBucketEntry: (rules, result, done) => done() },
+                rules: config.Rules,
+                value: result.value,
+            }, cb);
+        });
+    }
+
+    /**
+     * Set up the backbeat producer with the given topic.
+     * @param {String} topic - - Kafka topic to write to
+     * @param {Function} cb - The callback to call
+     * @return {undefined}
+     */
+    _setupProducer(topic, cb) {
+        const producer = new BackbeatProducer({
+            zookeeper: {
+                connectionString: this._zkConfig.connectionString,
+            },
+            topic,
+        });
+        producer.once('error', cb);
+        producer.once('ready', () => {
+            producer.removeAllListeners('error');
+            producer.on('error', err => {
+                this._log.error('error from backbeat producer', {
+                    topic,
+                    error: err,
+                });
+            });
+            return cb(null, producer);
+        });
+    }
+
+    /**
+     * Set up the lifecycle consumer.
+     * @return {undefined}
+     */
+    _setupConsumer() {
+        const consumer = new BackbeatConsumer({
+            zookeeper: {
+                connectionString: this._zkConfig.connectionString,
+            },
+            topic: this._lcConfig.bucketTasksTopic,
+            groupId: this._lcConfig.producer.groupId,
+            concurrency: this._lcConfig.producer.concurrency,
+            queueProcessor: this._processBucketEntry.bind(this),
+            fetchMaxBytes: CONSUMER_FETCH_MAX_BYTES,
+        });
+        consumer.on('error', () => {});
+        consumer.subscribe();
+    }
+
+    /**
+     * Set up the vault client cache for making requests to vault.
+     * @return {undefined}
+     */
+    _setupVaultClientCache() {
+        const { vault } = this._authConfig;
+        const { host, port, adminPort, adminCredentialsFile } = vault;
```

Comment:
just a quick note, `adminCredentialsFile` will need to be added to config files and a joi required field

Reply:
Was relying on this addition: https://github.com/scality/backbeat/blob/ft/S3C-1154-lifecycle/conf/config.joi.js#L31 and the error thrown if we cannot correctly parse credentials from the file. I guess we should also check for the existence of the `adminCredentialsFile` field, or else change `.optional()` to `.required()` in the joi validation.